### DISCUSSION TITLE
Add COSXL support

### DISCUSSION THREAD

**Discussion by ivaxsirc** [Panchovix/stable-diffusion-webui-reForge#14] [Ideas]
Could you add COSXL support?

Thanks

**Comment by Panchovix** [comment]
CosXL should work as is suported

https://github.com/Panchovix/stable-diffusion-webui-reForge/commit/77bdb9208d019e562f9f629647356dca2b2d5ef1

**Reply by ivaxsirc** [reply]
no, appears this error, RuntimeError: Given groups=1, weight of size [320, 8, 3, 3], expected input[2, 4, 64, 64] to have 8 channels, but got 4 channels instead
with the model https://civitai.com/models/397741/robmix-cosxl-edit
this model is Cosxl Pix2Pix

thanks

**Reply by Panchovix** [reply]
I see, can you give me the complete error of the console?

**Reply by Panchovix** [reply]
Also, I did some commits here https://github.com/Panchovix/stable-diffusion-webui-reForge/commit/ce3770d1d2b6a55cc429901113b7a8e7944d006b

It was missing some pix2pix models, can you try again?

**Reply by ivaxsirc** [reply]
Could you fix the problem?

another question when i'm doing a git pull appears this, something is wrong?
thks
O:\reforge\stable-diffusion-webui-reForge>git pull
remote: Enumerating objects: 73, done.
remote: Counting objects: 100% (67/67), done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 43 (delta 29), reused 29 (delta 18), pack-reused 0
Unpacking objects: 100% (43/43), 6.81 KiB | 43.00 KiB/s, done.
From https://github.com/Panchovix/stable-diffusion-webui-reForge
   c8799a3c..93a40cfa  dev_upstream -> origin/dev_upstream
   1020e8e7..0d7cdd3b  main         -> origin/main
Your configuration specifies to merge with the ref 'refs/heads/dev_upstream_a1111'
from the remote, but no such ref was fetched.

**Reply by Panchovix** [reply]
I haven't tinkered with CosXL yet, sorry.

For that git issue, do

```
git fetch remote

git checkout main
```

And it should work.

dev_upstream branch maybe has mroe CosXL compability, but I haven't tested.

**Comment by ivaxsirc** [comment]
Not works, this is the message, thks

venv "O:\reforge\stable-diffusion-webui-reForge\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f0.0.18v1.10.0RC-latest-566-g8cfb093f
Commit hash: 8cfb093fafc677bbe54407fa12cadb00ab32b8ed
Installing requirements
CUDA 12.1
Installing diffusers@wuerstchen-v3
Installing forge_legacy_preprocessor requirement: changing opencv-python version from 4.10.0.84 to 4.8.0
Installing sd-forge-controlnet requirement: changing opencv-python version from 4.10.0.84 to 4.8.0
Launching Web UI with arguments: --ckpt-dir 'F:\webui\stable-diffusion-webui\models\Stable-diffusion' --lora-dir F:/webui/stable-diffusion-webui/models/Lora --vae-dir F:/webui/stable-diffusion-webui/vae --show-controlnet-example
Total VRAM 24564 MB, total RAM 65462 MB
Total VRAM 24564 MB, total RAM 65462 MB
pytorch version: 2.1.2+cu121
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4090 : native
CUDA Stream Activated:  False
Using pytorch cross attention
*** Extension "sd-webui-cn-sam-preprocessor" requires "sd-webui-controlnet" which is disabled.
*** Extension "sd-webui-lama-cleaner-masked-content" requires "sd-webui-controlnet" which is disabled.
ControlNet preprocessor location: O:\reforge\stable-diffusion-webui-reForge\models\ControlNetPreprocessor
[-] ADetailer initialized. version: 24.6.0, num models: 10
*** Error loading script: cn_sam_preprocessor.py
    Traceback (most recent call last):
      File "O:\reforge\stable-diffusion-webui-reForge\modules\scripts.py", line 581, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "O:\reforge\stable-diffusion-webui-reForge\modules\script_loading.py", line 12, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "O:\reforge\stable-diffusion-webui-reForge\extensions\sd-webui-cn-sam-preprocessor\scripts\cn_sam_preprocessor.py", line 3, in <module>
        from scripts.supported_preprocessor import Preprocessor
    ModuleNotFoundError: No module named 'scripts.supported_preprocessor'

---
[sdwi2iextender] Developper warning:
[sdwi2iextender] ./modules/img2img.py is being recompiled at run time with a patch. Your debugger will not work in this file.
[sdwi2iextender] If you need debug tools in this file, disable all extensions that use the sdwi2iextender library.
[sdwi2iextender] This patch is temporary and will be removed when v1.9 will be released.
2024-07-15 07:58:00,782 - model_patcher_hook.py - INFO - __init__ hooks applied
__init__ hooks applied
2024-07-15 07:58:00,783 - model_patcher_hook.py - INFO - close hooks applied
close hooks applied
2024-07-15 07:58:00,784 - model_patcher_hook.py - INFO - sample hooks applied
sample hooks applied
07:58:01 - ReActor - STATUS - Running v0.7.1-a1 on Device: CUDA
Loading model cosXL\robmixCosxlEdit_cosxlZenithV20.safetensors [36809ed9f7] (1 out of 1)
Loading weights [36809ed9f7] from F:\webui\stable-diffusion-webui\models\Stable-diffusion\cosXL\robmixCosxlEdit_cosxlZenithV20.safetensors
*** module 'lib_controlnet.external_code' has no attribute 'get_models' ***
    Traceback (most recent call last):
      File "O:\reforge\stable-diffusion-webui-reForge\extensions\sd-webui-replacer\replacer\extensions\controlnet.py", line 213, in getInpaintModels
        models = external_code.get_models()
    AttributeError: module 'lib_controlnet.external_code' has no attribute 'get_models'

---
*** module 'lib_controlnet.external_code' has no attribute 'get_models' ***
    Traceback (most recent call last):
      File "O:\reforge\stable-diffusion-webui-reForge\extensions\sd-webui-replacer\replacer\extensions\controlnet.py", line 213, in getInpaintModels
        models = external_code.get_models()
    AttributeError: module 'lib_controlnet.external_code' has no attribute 'get_models'

---
First weight key found: conditioner.embedders.0.transformer.text_model.embeddings.position_embedding.weight
model_type V_PREDICTION_EDM
UNet ADM Dimension 2816
2024-07-15 07:58:03,312 - ControlNet - INFO - ControlNet UI callback registered.
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 64.1s (prepare environment: 46.3s, import torch: 5.3s, import gradio: 1.0s, setup paths: 1.1s, initialize shared: 0.2s, other imports: 0.7s, load scripts: 7.1s, scripts before_ui_callback: 0.1s, create ui: 1.4s, gradio launch: 0.5s, app_started_callback: 0.1s).
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
clip missing: ['clip_l.text_projection', 'clip_l.logit_scale', 'clip_g.transformer.text_model.embeddings.position_ids']
left over keys: dict_keys(['edm_vpred.sigma_max', 'edm_vpred.sigma_min'])
2024-07-15 07:58:09,258 - model_patcher_hook.py - INFO - Init p.model_patcher.
Init p.model_patcher.
2024-07-15 07:58:09,274 - model_patcher_hook.py - INFO - Init p.hr_model_patcher.
Init p.hr_model_patcher.
Requested to load SDXLClipModel
Loading 1 new model
Model loaded in 7.1s (load weights from disk: 0.4s, forge load real models: 6.2s, calculate empty prompt: 0.5s).
Starting job task(c1pujc0uompwqdj)
2024-07-15 07:58:45,197 - model_patcher_hook.py - INFO - Init p.model_patcher.
Init p.model_patcher.
Requested to load AutoencoderKL
Loading 1 new model
Requested to load SDXL
Loading 1 new model
2024-07-15 07:58:46,527 - model_patcher_hook.py - INFO - Patch model_patcher of StableDiffusionProcessingImg2Img.
Patch model_patcher of StableDiffusionProcessingImg2Img.
  0%|                                                                                           | 0/16 [00:00<?, ?it/s]
2024-07-15 07:58:46,815 - model_patcher_hook.py - INFO - Unpatch model_patcher of StableDiffusionProcessingImg2Img.
Unpatch model_patcher of StableDiffusionProcessingImg2Img.
2024-07-15 07:58:46,816 - model_patcher_hook.py - INFO - Close p.model_patcher.
Close p.model_patcher.
*** Error completing request
*** Arguments: ('task(c1pujc0uompwqdj)', <gradio.routes.Request object at 0x00000209814CC0A0>, 0, 'a man with a dog', '', [], <PIL.Image.Image image mode=RGBA size=1024x1024 at 0x209814DB5E0>, None, None, None, None, None, None, 4, 0, 1, 1, 1, 7, 1.5, 0.75, 0.0, 512, 512, 1, 0, 0, 32, 0, '', '', '', [], False, [], '', 0, False, 1, 0.5, 4, 0, 0.5, 2, 20, 'DPM++ 2M', 'Automatic', False, '', 0.8, -1, False, -1, 0, 0, 0, None, 0.5, False, False, {'ad_model': 'face_yolov8n.pt', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'ad_model': 'None', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'ad_model': 'None', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'ad_model': 'None', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'enabled': False, 'model_type': 'IC-Light.SD15.FC', 'input_fg': None, 'uploaded_bg': None, 'bg_source_fc': 'Custom LightMap', 'bg_source_fbc': 'Use Background Image', 'remove_bg': True, 'reinforce_fg': False, 'detail_transfer': False, 'detail_transfer_use_raw_input': False, 'detail_transfer_blur_radius': 5}, False, False, 0, None, [], 0, False, [], [], False, 0, 1, False, False, 0, None, [], -2, False, [], False, 0, None, None, 0.0, None, None, None, False, '0', '0', 'inswapper_128.onnx', 'CodeFormer', 1, True, 'None', 1, 1, False, True, 1, 0, 0, False, 0.5, True, False, 'CUDA', False, 0, 'None', '', None, False, False, 0.5, 0, 'tab_single', False, True, True, 'cosXL\\robmixCosxlEdit_cosxlZenithV20.safetensors [36809ed9f7]', True, True, False, '', '', '', '', 'None', -1, 'DPM++ 2M SDE', 'Automatic', 20, 0.3, 35, 4, 1280, '', 'GroundingDINO_SwinT_OGC (694MB)', 5.5, 1, 40, 0, 512, 512, 1, 1, 0, ['script'], False, False, 'cosXL\\robmixCosxlEdit_cosxlZenithV20.safetensors [36809ed9f7]', 'Random', ['Draw mask'], None, True, ['Draw mask'], None, True, None, 'ESRGAN_4x', 1, False, False, False, '-', -1, 0, 'ESRGAN_4x', 4, 'Use same sampler', 'Use same scheduler', 0.35, 1, '', 1800, 'Lanczos', True, True, 5, '', '', 'Use same checkpoint', 250, 2, True, 'Same', ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=None, batch_mask_gallery=None, multi_inputs_gallery=None, generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1.0, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0.0, guidance_end=1.0, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=None, batch_mask_gallery=None, multi_inputs_gallery=None, generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1.0, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0.0, guidance_end=1.0, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=None, batch_mask_gallery=None, multi_inputs_gallery=None, generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1.0, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0.0, guidance_end=1.0, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), False, 1, 0.5, 4, 0, 0.5, 2, ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, save_detected_map=True), False, 7, 1, 'Constant', 0, 'Constant', 0, 1, 'enable', 'MEAN', 'AD', 1, False, 1.01, 1.02, 0.99, 0.95, False, 0.5, 2, False, 256, 2, 0, False, False, 3, 2, 0, 0.35, True, 'bicubic', 'bicubic', False, 0, 'anisotropic', 0, 'reinhard', 100, 0, 'subtract', 0, 0, 'gaussian', 'add', 0, 100, 127, 0, 'hard_clamp', 5, 0, 'None', 'None', False, 'MultiDiffusion', 768, 768, 64, 4, False, False, False, '* `CFG Scale` should be 2 or lower.', True, True, '', '', True, 50, True, 1, 0, False, 4, 0.5, 'Linear', 'None', '<p style="margin-bottom:0.75em">Recommended settings: Sampling Steps: 80-100, Sampler: Euler a, Denoising strength: 0.8</p>', 128, 8, ['left', 'right', 'up', 'down'], 1, 0.05, 128, 4, 0, ['left', 'right', 'up', 'down'], False, False, 'positive', 'comma', 0, False, False, 'start', '', '<p style="margin-bottom:0.75em">Will upscale the image by the selected scale factor; use width and height sliders to set tile size</p>', 64, 0, 2, 1, '', [], 0, '', [], 0, '', [], True, False, False, False, False, False, False, 0, False) {}
    Traceback (most recent call last):
      File "O:\reforge\stable-diffusion-webui-reForge\modules\call_queue.py", line 74, in f
        res = list(func(*args, **kwargs))
      File "O:\reforge\stable-diffusion-webui-reForge\modules\call_queue.py", line 53, in f
        res = func(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\modules\call_queue.py", line 37, in f
        res = func(*args, **kwargs)
      File "<string>", line 234, in img2img
      File "O:\reforge\stable-diffusion-webui-reForge\modules\processing.py", line 815, in process_images
        res = process_images_inner(p)
      File "O:\reforge\stable-diffusion-webui-reForge\modules\processing.py", line 988, in process_images_inner
        samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
      File "O:\reforge\stable-diffusion-webui-reForge\modules\processing.py", line 1816, in sample
        samples = self.sampler.sample_img2img(self, self.init_latent, x, conditioning, unconditional_conditioning, image_conditioning=self.image_conditioning)
      File "O:\reforge\stable-diffusion-webui-reForge\modules\sd_samplers_kdiffusion.py", line 182, in sample_img2img
        samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
      File "O:\reforge\stable-diffusion-webui-reForge\extensions\sd-webui-model-patcher\scripts\model_patcher_hook.py", line 100, in wrapped_sample_func
        return func(self, *args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\modules\sd_samplers_common.py", line 274, in launch_sampling
        return func()
      File "O:\reforge\stable-diffusion-webui-reForge\modules\sd_samplers_kdiffusion.py", line 182, in <lambda>
        samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\repositories\k-diffusion\k_diffusion\sampling.py", line 594, in sample_dpmpp_2m
        denoised = model(x, sigmas[i] * s_in, **extra_args)
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
        return self._call_impl(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
        return forward_call(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\modules\sd_samplers_cfg_denoiser.py", line 369, in forward
        denoised = sampling_function(model, x, sigma, uncond_patched, cond_patched, cond_scale, model_options, seed)
      File "O:\reforge\stable-diffusion-webui-reForge\ldm_patched\modules\samplers.py", line 290, in sampling_function
        cond_pred, uncond_pred = calc_cond_uncond_batch(model, cond, uncond_, x, timestep, model_options)
      File "O:\reforge\stable-diffusion-webui-reForge\ldm_patched\modules\samplers.py", line 259, in calc_cond_uncond_batch
        output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)
      File "O:\reforge\stable-diffusion-webui-reForge\ldm_patched\modules\model_base.py", line 96, in apply_model
        model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds).float()
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
        return self._call_impl(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
        return forward_call(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\ldm_patched\ldm\modules\diffusionmodules\openaimodel.py", line 886, in forward
        h = forward_timestep_embed(module, h, emb, context, transformer_options, time_context=time_context, num_video_frames=num_video_frames, image_only_indicator=image_only_indicator)
      File "O:\reforge\stable-diffusion-webui-reForge\ldm_patched\ldm\modules\diffusionmodules\openaimodel.py", line 61, in forward_timestep_embed
        x = layer(x)
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
        return self._call_impl(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
        return forward_call(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\ldm_patched\modules\ops.py", line 114, in forward
        return super().forward(*args, **kwargs)
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\conv.py", line 460, in forward
        return self._conv_forward(input, self.weight, self.bias)
      File "O:\reforge\stable-diffusion-webui-reForge\venv\lib\site-packages\torch\nn\modules\conv.py", line 456, in _conv_forward
        return F.conv2d(input, weight, bias, self.stride,
    RuntimeError: Given groups=1, weight of size [320, 8, 3, 3], expected input[2, 4, 64, 64] to have 8 channels, but got 4 channels instead

---
HTTP Request: POST http://127.0.0.1:7860/api/predict "HTTP/1.1 200 OK"
HTTP Request: POST http://127.0.0.1:7860/reset "HTTP/1.1 200 OK"

**Comment by ivaxsirc** [comment]
ok, no problem, if you need more information or testing tell me
 Thanks for your work

Missatge de Panchovix ***@***.***> del dia dt., 16 de jul.
2024 a les 8:31:

**Comment by ivaxsirc** [comment]
It's solved the problem?

Thanks

**Reply by Panchovix** [reply]
Sorry, I haven't continued with CosXL yet because other updates. Will take a look when I have a chance.
